index.js

```diff
@@ -0,0 +1,159 @@
+    if (event.keyCode === 13) {
+     event.preventDefault();
+     submitWord(inputBar.value);//Submit the word to database
+     inputBar.value = "";//Clear the input bar
+    }
+});
+
+function retrieveWords(){//Call this function to fetch database words
+    return new Promise(async(resolve) => {//Return a promise when they are fetched
+        let wordsPromise = await fetch(url, {method: 'GET'});//Get the words object, if promise resolves...
+        let json = await wordsPromise.json();//Turn it into json, which gives a promise, once promise resolves...
+        if(json.isSubmitMode){//Check if submit mode is on
+            document.getElementById("submitModeButton").classList.add("greenButtonClass");//If yes, make submitmode button green
+        }
+        resolve(json);//Send back the json data from the api as promise is resolved.
+    });
+}
+
+function pickWord(){//When Pick Word button is clicked
+    if(dataObject.words.length == 0){//If words array is empty then print no words added
+        document.getElementById("word").innerHTML = "No Words Added";
+    }
+    else{//Otherwise, pick a random word from the words array or display SUBMIT MODE
+        if(dataObject.isSubmitMode){//If submit mode is on. Client can only submit and not see words
+            document.getElementById("word").innerHTML = "Submit Mode is on"
+        }
+        else{//If submit mode is off and array has words then display a random one
+            document.getElementById("word").innerHTML = dataObject.words[Math.floor(Math.random() * dataObject.words.length)]
+        }
+        
+    }
+}
+
+async function submitWord(word){//This function sends to database with a put method. If a boolean value is given, means to enable disable submit mode OR remove all words from database. If string, then submit new word
+    let currentDatabaseObj = await retrieveWords();//Must fetch current words from database, otherwise each client have different local databasese and they OVERWRITE the database with their old and not-updated local databases
+    dataObject = currentDatabaseObj;//Make local database updated with online database
+    let inputBarText;
+    if(typeof(word) != "boolean"){//If parameter is not a boolean, then we want to submit a word.
+        if(word == ""){//Make sure no empty strings come. If it does, just leave function, Mistake enter clicked while inputbar was focused
+            return;//Exit submitWord() if enter clicked by mistake on input bar
+        }
+        else{//If normal word
+            dataObject.words.push(word);//Add the word to the local database now, which is a object
+            stringJSON = JSON.stringify(dataObject);//Turn the json into string for the API
+            inputBarText = "Added Word";//Make bar say word that we added
+        }
+    }
+    else{//If we got a boolean, we either want to turn on submit mode, or clear the database
+        if(word == true){//Submit Mode
```

Comment:
I wouldn't use the same input for submitMode and also clearing the database or the same database string to decipher between inputs and modes